### PR TITLE
Student projects endpoint start

### LINKED ISSUE #18: Create API Contract
- [x] add draft of endpoints
- [x] json data samples
- [x] required fields
- [x] error handling

### PR DESCRIPTION
**Pull request by Gvieve**
### Add
* Poros
   * student projects - spec and poro to grab appropriate data for student projects endpoint 
   * projects - spec and poro to grab consolidate student_projects poro into single object for serializer for student projects endpoint 
### Update / Remove
* API contract
   * refined proposed endpoint routes. 
   * Added json example for student projects data response
   * added error handling section and samples
* Factories
   * updated associations on factories because they were creating duplicates
   * all model specs that needed changes due to changes to factories

Closes #18 
__WIP__ Student Projects Endpoint is still in progress. Need to create serializer and pull data/poro into controller.

### CODE REVIEW

File: API_contract.md [hunk 1]
```diff
@@ -4,10 +4,127 @@
...
+
+The request provides the projects for a valid matching student id and mod (sent as a query parameter).
+* __Required__
+  * Mod query parameter must be included and have an integer value between 0-4. If the mod query parameter is missing, or is an invalid value an error is sent (see [error handling](#error-handling)).
+* Data is returned in ascending order by project number.
+
+  Example json response
+
+  ```json
+  {
```
Comment (jequeza) [inline API_contract.md:28]: This data payload looks good so far!

File: app/models/project_feedback.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ class ProjectFeedback < ApplicationRecord
   belongs_to :user, foreign_key: :instructor_id
   belongs_to :student_project, foreign_key: :project_id
   belongs_to :rubric_template_category
+  has_one :rubric_category, through: :rubric_template_category
```
Comment (jequeza) [inline app/models/project_feedback.rb:5]: This sure was some fun troubleshooting session!

**Review by b-enji-cmd — APPROVED** [review]
LGTM! Awesome work on the testing!!!

File: app/models/student_project.rb [hunk 3]
```diff
@@ -4,4 +4,19 @@ class StudentProject < ApplicationRecord
...
+    select('student_projects.*,
+      project_templates.name,
+      project_templates.mod,
+      project_templates.program,
+      project_templates.project_number,
+      project_templates.project_type,
+      project_templates.is_final,
+      project_templates.rubric_template_id')
+    .joins(:project_template)
+    .where('student_projects.student_id = ?', id)
```
Comment (jequeza) [inline app/models/student_project.rb:18]: Not aware we could use 2 separate where statements. Thanks for this! 🤯

File: app/models/user_profile.rb [hunk 4]
```diff
@@ -4,8 +4,6 @@ class UserProfile < ApplicationRecord
             :last_name,
             :program,
             :current_mod,
-            :starting_cohort,
```
Comment (jequeza) [inline app/models/user_profile.rb:7]: Glad this edge case was found! An instructor may not have a cohort at all!

File: app/poros/student_projects.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+class StudentProjects
```
Comment (jequeza) [inline app/poros/student_projects.rb:1]: Loving the PORO

File: db/seeds.rb [hunk 6]
```diff
@@ -1,7 +1,64 @@
...
+RubricTemplate.destroy_all
+RubricCategory.destroy_all
+User.destroy_all
+
+#reset pk sequences back to 1
+ActiveRecord::Base.connection.tables.each do |t|
+  ActiveRecord::Base.connection.reset_pk_sequence!(t)
+end
+
+#seed data
```
Comment (jequeza) [inline db/seeds.rb:17]: FACTORY BOT FTW!

**Review by jequeza — APPROVED** [review]
Looks great DEV!

**Pull request merged**